File: backend/stock-market-backend/src/main/java/com/example/stock_market_backend/StockDataDTO.java
package com.example.stock_market_backend;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Map;

public class StockDataDTO {

    // Map representing the "Time Series (Daily)" data.
    @JsonProperty("Time Series (Daily)")
    private Map<String, Map<String, String>> timeSeries;

    // Getters and setters
    public Map<String, Map<String, String>> getTimeSeries() {
        return timeSeries;
    }

    public void setTimeSeries(Map<String, Map<String, String>> timeSeries) {
        this.timeSeries = timeSeries;
    }
}

File: frontend/src/app/StockDataDTO.ts
export interface StockDataDTO {
    "Time Series (Daily)"?: { 
      [date: string]: {
        "1. open": string;
        "2. high": string;
        "3. low": string;
        "4. close": string;
        "5. volume": string;
      };
    };
  }
  